src/OXRS_WT32.cpp

```diff
@@ -599,35 +616,47 @@ void OXRS_WT32::_initialiseClimateSensor(void)
 // get values from climate sensor, store local, publish /tele
 void OXRS_WT32::_updateClimateSensor(void)
 {
-  // Ignore if climate sensor not found or has been disabled
-  if (!_climateSensorFound || _climateUpdateMs == 0)
+  // Ignore if disabled
+  if (_climateUpdateMs == 0)
   {
     return;
   }
 
   // Check if we need to get new readings and publish
   if ((millis() - _lastClimateUpdate) > _climateUpdateMs)
   {
-    // Read values from onboard sensor
-    sht.read();
-
-    int temperature = round(sht.getTemperature() * 10);
-    int humidity = round(sht.getHumidity() * 10);
-    _temperature = (double)temperature / 10.0;
-    _humidity = (double)humidity / 10.0;
+    float tempESP;
+    StaticJsonDocument<128> json;
 
-    // Publish climate to mqtt
-    StaticJsonDocument<64> json;
-    json["temperature"] = _temperature;
-    json["humidity"] = _humidity;
-    publishTelemetry(json.as<JsonVariant>());
+    #if defined(CONFIG_IDF_TARGET_ESP32S3)
+    // read temperature from ESP chip
+    temp_sensor_read_celsius(&tempESP);
+    json["esp32Temp"] = round(tempESP);
+    #endif
 
-    // update screen
-    if (_onClimateUpdate)
+    if (_climateSensorFound)
     {
-      _onClimateUpdate();
+      // Read values from onboard sensor
+      sht.read();
+
+      int temperature = round(sht.getTemperature() * 10);
+      int humidity = round(sht.getHumidity() * 10);
+      _temperature = (double)temperature / 10.0;
+      _humidity = (double)humidity / 10.0;
+
+      json["temperature"] = _temperature;
+      json["humidity"] = _humidity;
+
+      // update screen
+      if (_onClimateUpdate)
+      {
+        _onClimateUpdate();
+      }
     }
 
+    // Publish climate to mqtt
+    publishTelemetry(json.as<JsonVariant>());
```

Comment:
Just checking, will `publishTelemetry()` check for an empty JSON object and ignore this if there are no sensors found? Or do you need to wrap this in a `json is not null` check?